File: relationship-manager/wealth-projection-service/src/main/java/com/bank/rm/projection/controller/ProjectionController.java
package com.bank.rm.projection.controller;

import com.bank.rm.common.dto.ApiResponse;
import com.bank.rm.projection.dto.ProjectionDtos.*;
import com.bank.rm.projection.service.WealthProjectionService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import java.util.UUID;

@RestController
@RequestMapping("/api/v1/projections")
public class ProjectionController {
    private final WealthProjectionService projectionService;

    public ProjectionController(WealthProjectionService projectionService) {
        this.projectionService = projectionService;
    }

    @PostMapping("/{customerId}")
    public ResponseEntity<ApiResponse<ProjectionResponse>> project(
            @PathVariable UUID customerId, @RequestBody ProjectionRequest request) {
        return ResponseEntity.ok(ApiResponse.ok(projectionService.project(customerId, request)));
    }
}

File: relationship-manager/wealth-projection-service/src/main/kotlin/com/bank/rm/projection/controller/ProjectionController.kt
package com.bank.rm.projection.controller

import com.bank.rm.common.dto.ApiResponse
import com.bank.rm.projection.dto.ProjectionRequest
import com.bank.rm.projection.dto.ProjectionResponse
import com.bank.rm.projection.service.WealthProjectionService
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*
import java.util.UUID

@RestController
@RequestMapping("/api/v1/projections")
class ProjectionController(private val projectionService: WealthProjectionService) {

    @PostMapping("/{customerId}")
    @ResponseStatus(HttpStatus.CREATED)
    fun project(
        @PathVariable customerId: UUID,
        @Valid @RequestBody request: ProjectionRequest
    ): ApiResponse<ProjectionResponse> =
        ApiResponse(success = true, data = projectionService.project(customerId, request))

    @GetMapping("/{customerId}/latest")
    fun getLatest(@PathVariable customerId: UUID): ApiResponse<ProjectionResponse?> =
        ApiResponse(success = true, data = projectionService.getLatestProjection(customerId))
}
